test/integration/smoke/test_storage_policy.py

```diff
@@ -0,0 +1,263 @@
+        cls.zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
+        cls._cleanup = []
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+        cls.network_offering = NetworkOffering.create(
+            cls.apiclient,
+            cls.testdata["l2-network_offering"],
+        )
+        cls.network_offering.update(cls.apiclient, state='Enabled')
+        cls.template = get_test_template(
+            cls.apiclient,
+            cls.zone.id,
+            cls.hypervisor,
+        )
+        cls.cleanup = [cls.network_offering]
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+
+        self.dbclient = self.testClient.getDbConnection()
+        self.hypervisor = self.testClient.getHypervisorInfo()
+        self.testdata["virtual_machine"]["zoneid"] = self.zone.id
+        self.testdata["virtual_machine"]["template"] = self.template.id
+        self.account = Account.create(
+            self.apiclient,
+            self.testdata["account"],
+            domainid=self.domain.id
+        )
+        self.cleanup = [self.account]
+        return
+
+    def tearDown(self):
+        try:
+            self.debug("Cleaning up the resources")
+            cleanup_resources(self.apiclient, self.cleanup)
+            self.debug("Cleanup complete!")
+        except Exception as e:
+            self.debug("Warning! Exception in tearDown: %s" % e)
+
+    def import_vmware_storage_policies(self, apiclient):
+        cmd = importVsphereStoragePolicies.importVsphereStoragePoliciesCmd()
+        cmd.zoneid = self.zone.id
+        return apiclient.importVsphereStoragePolicies(cmd)
+
+    def list_storage_policies(self, apiclient):
+        cmd = listVsphereStoragePolicies.listVsphereStoragePoliciesCmd()
+        cmd.zoneid = self.zone.id
+        return apiclient.listVsphereStoragePolicies(cmd)
+
+    def create_volume(self, apiclient):
+        cmd = create
+    @attr(
+        tags=[
+            "advanced",
+            "eip",
+            "advancedns",
+            "basic",
+            "sg"],
+        required_hardware="true")
+    def test_01_import_storage_policies(self):
+        """Test VMWare storage policies
+        """
+
+        # Validate the following:
+        # 1. Import VMWare storage policies - the command should return non-zero result
+        # 2. List current VMWare storage policies - the command should return non-zero result
+        # 3. Create service offering with first of the imported policies
+        # 4. Create disk offering with first of the imported policies
+        # 5. Create VM using the already created service offering
+        # 6. Create volume using the already created disk offering
+        # 7. Attach this volume to our VM
+        # 8. Detach the volume from our VM
```

Comment:
sensible comments, but I would like to see this be partitioned into methods of which the names reflect these comments. this method is long and thus rather hard to understand. also it is hard to check if it does what it says on the tin, certainly after a few rounds of maintenance.